### PR TITLE
Pset_Maintenance and Pset_Purchase

### PR DESCRIPTION
**Pull request by czapplitec**

### CODE REVIEW

File: IFC4x3/Properties/a/AssessmentFrequency_3Y5J0kLVfDHfMH8$QTwre7/DocProperty.xml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="AssessmentFrequency_3Y5J0kLVfDHfMH8QTwre7" Name="AssessmentFrequency" UniqueId="e215302e-55fa-4d46-9591-23f69deb5a07" PrimaryDataType="IfcFrequencyMeasure" />
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Properties/a/AssessmentFrequency_3Y5J0kLVfDHfMH8$QTwre7/DocProperty.xml]: I would use `IfcTimeMeasure` as FrequencyMeasure is for Hz and vibration values!

File: IFC4x3/Properties/p/PuttingIntoOperationDate_03cT7VFl92dw8n1GPuqXSv/DocProperty.xml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PuttingIntoOperationDate_03cT7VFl92dw8n1GPuqXSv" Name="PuttingIntoOperationDate" UniqueId="0399d1df-3ef2-429f-a231-050678d21739" PrimaryDataType="IfcDate" />
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Properties/p/PuttingIntoOperationDate_03cT7VFl92dw8n1GPuqXSv/DocProperty.xml]: use `PutIntoOperationDate` shorter and more grammatically correct

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Maintenance/Documentation.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Properties defining maintenance information for the element, asset or system.
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Maintenance/Documentation.md]:
```suggestion
Properties defining maintenance information for a element, asset or system.
```

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Purchase/Documentation.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Properties defining purchase information for the element.
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Purchase/Documentation.md]:
```suggestion
Properties defining purchase information for a element.
```

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Purchase/DocPropertySet.xml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_Purchase" UniqueId="6bf8a00d-dab0-4a42-8d32-461e590b9854" ApplicableType="IfcElement" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
+	<Properties>
+		<DocProperty xsi:nil="true" href="PurchasePrice_0ScQezQyP2duRs0dFE3Jzw" />
+		<DocProperty xsi:nil="true" href="Supplier_3wkQvqvdXEkRfl5wppCLK_" />
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Purchase/DocPropertySet.xml]: I find this PSet a bit troubling as the situation it is addressing is far more complex than just a price and supplier.

Consider researching and using IfcProjectOrder to properly portray and and exchange this information?

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Maintenance/DocPropertySet.xml [hunk 6]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_Maintenance" UniqueId="a5ccf5dc-9c2d-8761-9e7d-06206c294ae5" ApplicableType="IfcElement,IfcAsset,IfcSystem" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Maintenance/DocPropertySet.xml]: right this property set is a mashup of many distinct concepts, assessments, maintenance tasks & installation properties. we did alot of research with a few operators in P&W about how to manage maintenance data. it comes out at three levels: 

1. low level PSets like this example
2. structure group of PSets elaborating a maintenance strategy - P&W did this one!
3. Full blown use of Work Scheduling, tasks & work orders related to elements, systems & assets.

 first point is to remember the same property cannot be applied to an occurence multiple times unless in differently named property sets.

### Maintenance properties
MaintenanceDate, NextMaintenanceDate, MaintenanceType, and possibly DisassemblyDate are already covered by the combination of `Pset_MaintenanceStrategy` & `Pset_MaintenanceTriggerDuration` which is what this was designed for.

### Installation properties
these should be grouped into there own property set 

### Assessments
PSet_Condition is the primary property set for providing the latest condition attached to an element. but as the starting caveat this is overridden every time a new assessment is made. to keep track of assessments and there details, an wider work needs to be conducted around IfcTask and Inspection information.

### Repairs 
A similar latest repair property set like PSet_Condition might be useful to cover repair properties.

**Review by AlexBrad1eyCT — CHANGES_REQUESTED** [review]
Question: why do we need secondary tyoes on reference values??

This is going to need a full blown discussion at some point (and it aint going to be a short one!)

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Purchase/Documentation.md (hunk 4 above)
Comment (larswik) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Purchase/Documentation.md]:
```suggestion
Properties defining purchase information for an element.
```

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Purchase/DocPropertySet.xml (hunk 5 above)
Comment (larswik) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Purchase/DocPropertySet.xml]: Yes, this feels like a very generic kind of information which make me feel like we need to be careful. There are also things like IfcCostItem to look at maybe? Feels like this one is already covered.

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Maintenance/DocPropertySet.xml (hunk 6 above)
Comment (larswik) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Maintenance/DocPropertySet.xml]: Just want to point out that there are other properties of interest, such as serial numbers, batch numbers, article numbers, manufacturer, production year etc that are of interest for maintenance, but these are already available in Pset_ManufacturerOccurrence and Pset_ManufacturerTypeInformation. Therefore supporting Alex's comment to put the installation properties (InstallationDate, PutIntoOperationDate - what more?) in a separate more specific Pset_Installation*?

File: IFC4x3/Properties/a/AcceptanceDate_12dHQ2H3D5Vw6e1UM1sbnF/DocProperty.xml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="AcceptanceDate_12dHQ2H3D5Vw6e1UM1sbnF" Name="AcceptanceDate" UniqueId="429d1682-4433-457f-a1a8-05e581da5c4f" PrimaryDataType="IfcDate" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/a/AcceptanceDate_12dHQ2H3D5Vw6e1UM1sbnF/DocProperty.xml:2]: Seems like commencement date if it is modified a bit.

File: IFC4x3/Properties/a/AssessmentFrequency_3Y5J0kLVfDHfMH8$QTwre7/DocProperty.xml (hunk 1 above)
Comment (SergejMuhic) [reply IFC4x3/Properties/a/AssessmentFrequency_3Y5J0kLVfDHfMH8$QTwre7/DocProperty.xml]: I guess this depends on what information is expected. Number/time?

File: IFC4x3/Properties/a/AssessmentFrequency_3Y5J0kLVfDHfMH8$QTwre7/Documentation.md [hunk 8]
```diff
@@ -0,0 +1 @@
+Indicates how often the equipment is assessed, to have a clear estimation on its working state, based on which the maintenance staff can decide whether it requires maintenance or requires to be updated or replaced.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/a/AssessmentFrequency_3Y5J0kLVfDHfMH8$QTwre7/Documentation.md]:
```suggestion
Indicates how often the equipment is assessed to have a clear estimation on its working state. Based on this the maintenance staff can decide whether it requires maintenance or requires to be updated or replaced.
```

File: IFC4x3/Properties/d/DisassemblyDate_21F9eF2br0OPwbIT8YD7$C/Documentation.md [hunk 9]
```diff
@@ -0,0 +1 @@
+Date on which the element is removed.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/d/DisassemblyDate_21F9eF2br0OPwbIT8YD7$C/Documentation.md]: Removed?

File: IFC4x3/Properties/m/MaintenanceServiceCycle_01ShZOXG5BLO_Gv105Bz_e/DocProperty.xml [hunk 10]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="MaintenanceServiceCycle_01ShZOXG5BLO_Gv105Bz_e" Name="MaintenanceServiceCycle" UniqueId="0172b8d8-8501-4b55-8f90-e410052fdfa8" PrimaryDataType="IfcTimeMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/m/MaintenanceServiceCycle_01ShZOXG5BLO_Gv105Bz_e/DocProperty.xml]: Seems a nicer way of how to document cycles. Consider it at `AssesmentFrequency`.

File: IFC4x3/Properties/r/RepairingContent_2uB5y1E3H1zuDscjqI6Rjw/Documentation.md [hunk 11]
```diff
@@ -0,0 +1 @@
+Content of repair, reason and nature can be given, e.g. display faults, communication failure, display exchange.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/r/RepairingContent_2uB5y1E3H1zuDscjqI6Rjw/Documentation.md]:
```suggestion
Content of the property involves repair, reason and nature, e.g. display faults, communication failure, display exchange.
```

File: IFC4x3/Properties/r/RepairingDate_1fKpvOEozEKQHboTTf4_f7/Documentation.md [hunk 12]
```diff
@@ -0,0 +1 @@
+Date on which the last repair is done on the asset.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/r/RepairingDate_1fKpvOEozEKQHboTTf4_f7/Documentation.md]:
```suggestion
Date on which the last repair was done on the asset.
```

File: IFC4x3/Properties/s/StorageCondition_2fY6Ixwun9OeL0AdKAOGuo/Documentation.md [hunk 13]
```diff
@@ -0,0 +1 @@
+Indicates the storage condition of equipment
```
Comment (SergejMuhic) [inline IFC4x3/Properties/s/StorageCondition_2fY6Ixwun9OeL0AdKAOGuo/Documentation.md]:
```suggestion
Indicates the storage condition of equipment.
```

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Maintenance/DocPropertySet.xml (hunk 6 above)
Comment (SergejMuhic) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Maintenance/DocPropertySet.xml]: I have also looked at Pset_Condition and it should probably get a couple of these properties.

I do agree with the assessment above.

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Maintenance/Documentation.md (hunk 3 above)
Comment (SergejMuhic) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Maintenance/Documentation.md]:
```suggestion
Properties defining maintenance information for an element, asset or system.
```

Comment (SergejMuhic) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Maintenance/Documentation.md]: 😄

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Purchase/DocPropertySet.xml (hunk 5 above)
Comment (SergejMuhic) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Purchase/DocPropertySet.xml]: How about considering actually modelling cost items?

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Purchase/Documentation.md (hunk 4 above)
Comment (SergejMuhic) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_Purchase/Documentation.md]:
```suggestion
Properties defining purchase information for an element.
```

File: IFC4x3/Properties/a/AssessmentFrequency_3Y5J0kLVfDHfMH8$QTwre7/DocProperty.xml [hunk 14]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="AssessmentFrequency_3Y5J0kLVfDHfMH8QTwre7" Name="AssessmentFrequency" UniqueId="e215302e-55fa-4d46-9591-23f69deb5a07" PrimaryDataType="IfcTime" />
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Properties/a/AssessmentFrequency_3Y5J0kLVfDHfMH8$QTwre7/DocProperty.xml]: i think it should be IfcTimeMeasure: 

- IfcTime is a particular point in time
- IfcTimeMeasure like 6 months or 1 year

File: IFC4x3/Properties/r/RepairingDate_1fKpvOEozEKQHboTTf4_f7/DocProperty.xml [hunk 15]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="RepairingDate_1fKpvOEozEKQHboTTf4_f7" Name="RepairingDate" UniqueId="69533e58-3b2f-4e51-a465-c9d76913ea47" PrimaryDataType="IfcDate" />
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Properties/r/RepairingDate_1fKpvOEozEKQHboTTf4_f7/DocProperty.xml]: Try RepairDate

File: IFC4x3/Properties/r/RepairingContent_2uB5y1E3H1zuDscjqI6Rjw/Documentation.md (hunk 11 above)
Comment (AlexBrad1eyCT) [inline IFC4x3/Properties/r/RepairingContent_2uB5y1E3H1zuDscjqI6Rjw/Documentation.md]: RepairContent

**Review by AlexBrad1eyCT — CHANGES_REQUESTED** [review]
this looks acceptable, just a few naming suggestions, plus dont forget `PutIntoOperationDate`

File: IFC4x3/Properties/a/AcceptanceDate_12dHQ2H3D5Vw6e1UM1sbnF/DocProperty.xml (hunk 7 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Properties/a/AcceptanceDate_12dHQ2H3D5Vw6e1UM1sbnF/DocProperty.xml:2]: nah its different i more akind to the date it is approved and accepted by the client.

File: IFC4x3/Properties/a/AssessmentFrequency_3Y5J0kLVfDHfMH8$QTwre7/DocProperty.xml (hunk 14 above)
Comment (SergejMuhic) [reply IFC4x3/Properties/a/AssessmentFrequency_3Y5J0kLVfDHfMH8$QTwre7/DocProperty.xml]: Yup IfcTime is a particular time in a day.

File: IFC4x3/Properties/r/RepairingContent_2uB5y1E3H1zuDscjqI6Rjw/DocProperty.xml [hunk 16]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="RepairingContent_2uB5y1E3H1zuDscjqI6Rjw" Name="RepairingContent" UniqueId="b82c5f01-3834-41f7-8376-9add1219bb7a" PrimaryDataType="IfcText" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/r/RepairingContent_2uB5y1E3H1zuDscjqI6Rjw/DocProperty.xml]: RepairContent here.

**Review by AlexBrad1eyCT — APPROVED** [review]

File: IFC4x3/Properties/a/AssessmentFrequency_3Y5J0kLVfDHfMH8$QTwre7/DocProperty.xml (hunk 14 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Properties/a/AssessmentFrequency_3Y5J0kLVfDHfMH8$QTwre7/DocProperty.xml:2]:
```suggestion
<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="AssessmentFrequency_3Y5J0kLVfDHfMH8QTwre7" Name="AssessmentFrequency" UniqueId="e215302e-55fa-4d46-9591-23f69deb5a07" PrimaryDataType="IfcTimeMeasure" />
```
```suggestion
<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="AssessmentFrequency_3Y5J0kLVfDHfMH8QTwre7" Name="AssessmentFrequency" UniqueId="e215302e-55fa-4d46-9591-23f69deb5a07" PrimaryDataType="IfcTime" />
```

**Review by larswik — APPROVED** [review]

File: IFC4x3/Properties/a/AssessmentFrequency_3Y5J0kLVfDHfMH8$QTwre7/DocProperty.xml (hunk 14 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Properties/a/AssessmentFrequency_3Y5J0kLVfDHfMH8$QTwre7/DocProperty.xml]:
```suggestion
<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="AssessmentFrequency_3Y5J0kLVfDHfMH8QTwre7" Name="AssessmentFrequency" UniqueId="e215302e-55fa-4d46-9591-23f69deb5a07" PrimaryDataType="IfcTimeMeasure" />
```

**Review by SergejMuhic — APPROVED** [review]

**Pull request merged**